float64[3] fl_pos
float64[3] fr_pos
float64[3] rl_pos
float64[3] rr_pos
bool fl_set
bool fr_set
bool rl_set
bool rr_set